From fe25fadfe4ecae6600c81b1cab1c9e7f14dfe27d Mon Sep 17 00:00:00 2001
From: Martin Fick <mfick@codeaurora.org>
Date: Fri, 10 Jun 2011 17:44:50 -0600
Subject: [PATCH] Fix unified comment editing with single click on line numbers

Allow editing to happen also when line number is in first
or second column.  Add anchors for styling.

Change-Id: I35a6d4976c0a0887cf395074cd8b1a9ca8146886
---
 .../gerrit/client/patches/SideBySideTable.java     |    3 +++
 .../gerrit/client/patches/UnifiedDiffTable.java    |    9 +++++++--
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/SideBySideTable.java b/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/SideBySideTable.java
index 26e5dbd..ea48215 100644
--- a/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/SideBySideTable.java
+++ b/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/SideBySideTable.java
@@ -342,7 +342,10 @@ public class SideBySideTable extends AbstractPatchContentTable {
       final boolean fullBlock) {
     m.openTd();
     m.setStyleName(Gerrit.RESOURCES.css().lineNumber());
+    m.openAnchor();
+    m.setAttribute("href", "");
     m.append(lineNumberMinusOne + 1);
+    m.closeAnchor();
     m.closeTd();
 
     m.openTd();
diff --git a/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/UnifiedDiffTable.java b/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/UnifiedDiffTable.java
index b9e6afa..c700b4e 100644
--- a/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/UnifiedDiffTable.java
+++ b/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/UnifiedDiffTable.java
@@ -66,8 +66,10 @@ public class UnifiedDiffTable extends AbstractPatchContentTable {
 
   @Override
   protected void onCellSingleClick(int row, int column) {
-    if (column == 2) {
-      onCellDoubleClick(row, column);
+    if (column == 1 || column == 2) {
+      if (!"".equals(table.getText(row, column))) {
+        onCellDoubleClick(row, column);
+      }
     }
   }
 
@@ -381,7 +383,10 @@ public class UnifiedDiffTable extends AbstractPatchContentTable {
   private void appendLineNumber(final SafeHtmlBuilder m, final int idx) {
     m.openTd();
     m.setStyleName(Gerrit.RESOURCES.css().lineNumber());
+    m.openAnchor();
+    m.setAttribute("href", "");
     m.append(idx + 1);
+    m.closeAnchor();
     m.closeTd();
   }
 }
-- 
1.7.4.1

